pallets/storage-provider/benchmarks/src/benchmarking.rs

```diff
@@ -271,3 +358,15 @@ where
     pallet_balances::Pallet::<T>::make_free_balance_be(&account.0, balance.into());
     account
 }
+
+/// Asserts that the result was Ok, and additionally decodes the error if it was a GeneralPalletError
+/// to get the inner error as a user-friendly string.
+fn assert_ok_sp(res: Result<(), DispatchError>) {
+    if let Err(sp_runtime::DispatchError::Module(module)) = &res {
+        if let Some("GeneralPalletError") = module.message {
+            let variant = GeneralPalletError::decode(&mut [module.error[1]].as_slice()).unwrap();
+            panic!("{variant:#?}");
+        }
```

Comment:
```suggestion
            if let Ok(variant) = GeneralPalletError::decode(&mut [module.error[1]].as_slice()) {
            panic!("{variant:#?}");
        }
```
Would this work, or is the `&str` check required for the decoding?

Reply:
I think that _could_ decode without an error even if it wasn't a `GeneralPalletError`.

Reply:
That was what I was afraid of.